src/maxminddb.c

```diff
@@ -816,6 +823,8 @@ LOCAL record_info_s record_info_for_database(MMDB_s *mmdb)
         record_info.left_record_getter = &get_uint32;
         record_info.right_record_getter = &get_uint32;
         record_info.right_record_offset = 4;
+    } else {
+        assert(false);
```

Comment:
This is appropriate as an assert as we already checked that the record_size from the database was 24, 28, or 32. This should only happen if there was a programming error on our part.